### PR TITLE
Avoid using rand_os crate (fixes warnings)

### PR DESCRIPTION
**Pull request by NikVolf**

### CODE REVIEW

**Comment by str4d** [comment]
`rand 0.7` depends on `rand_core 0.5`, so I don't understand how this is an update. In any case, we made the explicit decision in #91 to prefer `rand_core` over `rand` in order to reduce the number of dependencies (we only depend on the `rand` crate in a couple of places, and plan to refactor those to remove the `rand` dependency entirely).

The warning fixes are definitely welcome :slightly_smiling_face:

**Comment by NikVolf** [comment]
You can’t depend on rand_core and get no deprecated warnings anymore

**Comment by NikVolf** [comment]
Actually, seems that only rand_os::* produces warnings, I can't update just that

**Comment by str4d** [comment]
Reading the warning:
```
warning: use of deprecated item 'rand_os::OsRng': OsRng is now provided by rand_core and rand
```

I would prefer that we switch from `rand_os` to `rand_core` instead of `rand`. That should also be a smaller change, as almost all crates already specify `rand_core` as an explicit dependency.

**Comment by NikVolf** [comment]
@str4d 

Warning is misleading, there is no mention of OsRng is rand_core docs, and module `os` in this crate is private

File: zcash_proofs/Cargo.toml [hunk 1]
```diff
@@ -13,11 +13,10 @@ byteorder = "1"
 directories = { version = "1", optional = true }
 ff = { path = "../ff" }
 pairing = { path = "../pairing" }
-rand_os = "0.2"
 zcash_primitives = { path = "../zcash_primitives" }
+rand_core = "0.5"
```
Comment (str4d) [inline zcash_proofs/Cargo.toml]: Nit: alphabetize dependencies.

**Comment by jimpo** [comment]
This broke my project which uses the `zcash_primitives` crate.

`rand_core` 0.5.1 exports `OsRng` in the root, but 0.5.0 does not. The `Cargo.toml` for this project only specifies `0.5`.

**Comment by NikVolf** [comment]
@jimpo How do you reference zcash_primitives in your project?

**Comment by jimpo** [comment]
`zcash_primitives = { git = "https://github.com/zcash/librustzcash" }`

Putting `rand_core = "0.5.1"` in my `Cargo.toml` fixes it, but I shouldn't have to do that.

**Comment by NikVolf** [comment]
I think it's just your local Cargo.lock can be updated with `cargo update -p rand_core`, no need to pin rand_core, it can get bumped again at some point

**Pull request merged**